<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Karissa Barthelson" />

<meta name="date" content="2023-08-23" />

<title>exp1.2_meltCurveGenotyping</title>

<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">2023_MPSIII_salbutamol</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">exp1.2_meltCurveGenotyping</h1>
<h4 class="author">Karissa Barthelson</h4>
<h4 class="date">2023-08-23</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-08-23
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>2023_MPSIII_salbutamol/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has staged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230614code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20230614)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230614code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230614)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomkarissab2023MPSIIIsalbutamoltree8381757306fbe44e1857d25b4f2bd04abd181cb8targetblank8381757a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/karissa-b/2023_MPSIII_salbutamol/tree/8381757306fbe44e1857d25b4f2bd04abd181cb8" target="_blank">8381757</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomkarissab2023MPSIIIsalbutamoltree8381757306fbe44e1857d25b4f2bd04abd181cb8targetblank8381757a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/karissa-b/2023_MPSIII_salbutamol/tree/8381757306fbe44e1857d25b4f2bd04abd181cb8" target="_blank">8381757</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    code/experiment 1.2/
    Ignored:    data/.DS_Store
    Ignored:    data/exp1.2/.DS_Store
    Ignored:    data/exp1.2/processed_data/
    Ignored:    data/exp1.2/raw_data/.DS_Store
    Ignored:    data/exp1.2/videos/
    Ignored:    data/exp1/.DS_Store
    Ignored:    data/exp1/processed_data/.DS_Store
    Ignored:    data/exp1/raw_data/.DS_Store
    Ignored:    data/exp1/videos/
    Ignored:    data/pilot/.DS_Store
    Ignored:    data/pilot/14 days/.DS_Store
    Ignored:    data/pilot/14 days/raw_data/.DS_Store
    Ignored:    data/pilot/7 days/.DS_Store
    Ignored:    data/pilot/7 days/raw_data/.DS_Store
    Ignored:    data/pilot/7 days/raw_data/distances/.DS_Store
    Ignored:    data/pilot/7 days/tracking_videos/

Untracked files:
    Untracked:  data/exp1.2/genotyping_data/meltcurveWithGenotypes.rds
    Untracked:  data/exp1.2/metadata_withGenotype.rds

Unstaged changes:
    Modified:   analysis/exp1.2_meltCurveGenotyping.rmd

Staged changes:
    Modified:   .gitignore
    New:        analysis/exp1.2_meltCurveGenotyping.rmd
    New:        analysis/fullexp1.rmd
    Modified:   analysis/pilot.rmd
    New:        analysis/piulot14days.rmd
    New:        code/experiment 1/distances_exp1.R
    New:        code/experiment 1/exp1_genotypingMeltCurve.R
    New:        code/experiment 1/tetras_preprocessing.R
    Modified:   code/pilot_distances_preprocessing.R
    New:        code/pilot_tetras_14d_preprocessing.R
    Modified:   code/pilot_tetras_preprocessing.R
    New:        data/exp1.2/2023_Aug18_salbutamolexp1_2_meta.xlsx
    New:        data/exp1.2/genotyping_data/KB_20230821_salbutamolExp1_2_nagluhomxhet7m_genotyping.eds
    New:        data/exp1.2/genotyping_data/KB_20230821_salbutamolExp1_2_nagluhomxhet7m_genotyping_amplificationdata.txt
    New:        data/exp1.2/genotyping_data/KB_20230821_salbutamolExp1_2_nagluhomxhet7m_genotyping_meltregionderivativedata.txt
    New:        data/exp1.2/genotyping_data/KB_20230821_salbutamolExp1_2_nagluhomxhet7m_genotyping_meltregionnormalizeddata.txt
    New:        data/exp1.2/genotyping_data/KB_20230821_salbutamolExp1_2_nagluhomxhet7m_genotyping_meltregiontemperaturedata.txt
    New:        data/exp1.2/genotyping_data/KB_20230821_salbutamolExp1_2_nagluhomxhet7m_genotyping_results.txt
    New:        data/exp1.2/genotyping_data/KB_20230821_salbutamolExp1_2_nagluhomxhet7m_genotyping_samplesetup.txt
    New:        data/exp1.2/raw_data/distances/ymaze_inLT-20230818T102151-data_totals.csv
    New:        data/exp1.2/raw_data/distances/ymaze_inLT-20230818T104226-data_totals.csv
    New:        data/exp1.2/raw_data/distances/ymaze_inLT-20230818T115513-data_totals.csv
    New:        data/exp1.2/raw_data/distances/ymaze_inLT-20230818T130757-data_totals.csv
    New:        data/exp1.2/raw_data/distances/ymaze_inLT-20230818T141842-data_totals.csv
    New:        data/exp1.2/raw_data/distances/ymaze_inLT-20230818T153027-data_totals.csv
    New:        data/exp1.2/raw_data/distances/ymaze_inLT-20230818T164048-data_totals.csv
    New:        data/exp1.2/raw_data/distances/ymazeinLT-20230818T095428-data_totals.csv
    New:        data/exp1.2/raw_data/distances/ymazeinLT-20230818T095643-data_totals.csv
    New:        data/exp1.2/raw_data/distances/ymazeinLT-20230818T104449-data_totals.csv
    New:        data/exp1.2/raw_data/distances/ymazeinLT-20230818T115820-data_totals.csv
    New:        data/exp1.2/raw_data/distances/ymazeinLT-20230818T131032-data_totals.csv
    New:        data/exp1.2/raw_data/distances/ymazeinLT-20230818T142117-data_totals.csv
    New:        data/exp1.2/raw_data/distances/ymazeinLT-20230818T153318-data_totals.csv
    New:        data/exp1.2/raw_data/distances/ymazeinLT-20230818T164213-data_totals.csv
    New:        data/exp1.2/raw_data/distances/ymazeinLT-20230818T164447-data_totals.csv
    New:        data/exp1.2/raw_data/distances/ymazeinLT-20230818T164456-data_totals.csv
    New:        data/exp1.2/raw_data/zones/ymaze8-20230818T095443.csv
    New:        data/exp1.2/raw_data/zones/ymaze_inLT-20230818T102150.csv
    New:        data/exp1.2/raw_data/zones/ymaze_inLT-20230818T104225.csv
    New:        data/exp1.2/raw_data/zones/ymaze_inLT-20230818T115512.csv
    New:        data/exp1.2/raw_data/zones/ymaze_inLT-20230818T130756.csv
    New:        data/exp1.2/raw_data/zones/ymaze_inLT-20230818T141841.csv
    New:        data/exp1.2/raw_data/zones/ymaze_inLT-20230818T153026.csv
    New:        data/exp1.2/raw_data/zones/ymaze_inLT-20230818T164047.csv
    New:        data/exp1.2/raw_data/zones/ymazeinLT-20230818T095427.csv
    New:        data/exp1.2/raw_data/zones/ymazeinLT-20230818T095643.csv
    New:        data/exp1.2/raw_data/zones/ymazeinLT-20230818T104448.csv
    New:        data/exp1.2/raw_data/zones/ymazeinLT-20230818T115819.csv
    New:        data/exp1.2/raw_data/zones/ymazeinLT-20230818T131031.csv
    New:        data/exp1.2/raw_data/zones/ymazeinLT-20230818T142116.csv
    New:        data/exp1.2/raw_data/zones/ymazeinLT-20230818T153317.csv
    New:        data/exp1.2/raw_data/zones/ymazeinLT-20230818T164213.csv
    New:        data/exp1.2/raw_data/zones/ymazeinLT-20230818T164446.csv
    New:        data/exp1.2/raw_data/zones/ymazeinLT-20230818T164456.csv
    New:        data/exp1/2023_June30_salbutamolexp1_meta.xlsx
    New:        data/exp1/2023_June30_salbutamolexp1_meta.xltx
    New:        data/exp1/genotyping_data/2023_0708_KB_melt_curve_genotyping_salbutamol_full_exp1_meltregionderivativedata.xls
    New:        data/exp1/genotyping_data/2023_0708_KB_melt_curve_genotyping_salbutamol_full_exp1_meltregiontemperaturedata.xls
    New:        data/exp1/genotyping_data/2023_0708_KB_melt_curve_genotyping_salbutamol_full_exp1_results.xls
    New:        data/exp1/genotyping_data/2023_0708_KB_melt_curve_genotyping_salbutamol_full_exp1_samplesetup.xls
    New:        data/exp1/genotyping_data/genotypninResults.csv
    New:        data/exp1/genotyping_data/meltcurveWithGenotypes.rds
    New:        data/exp1/metadata_withGenotype.rds
    New:        data/exp1/processed_data/distanceDataexp1.rds
    New:        data/exp1/processed_data/final_output.csv
    New:        data/exp1/processed_data/tidy_data.csv
    New:        data/exp1/processed_data/time_in_zone.csv
    New:        data/exp1/raw_data/distances/ymaze_inLT-20230707T103034-data_totals.csv
    New:        data/exp1/raw_data/distances/ymaze_inLT-20230707T114210-data_totals.csv
    New:        data/exp1/raw_data/distances/ymaze_inLT-20230707T125228-data_totals.csv
    New:        data/exp1/raw_data/distances/ymaze_inLT-20230707T140117-data_totals.csv
    New:        data/exp1/raw_data/distances/ymaze_inLT-20230707T151154-data_totals.csv
    New:        data/exp1/raw_data/distances/ymaze_inLT-20230707T162410-data_totals.csv
    New:        data/exp1/raw_data/distances/ymazeinLT-20230707T103252-data_totals.csv
    New:        data/exp1/raw_data/distances/ymazeinLT-20230707T114444-data_totals.csv
    New:        data/exp1/raw_data/distances/ymazeinLT-20230707T125502-data_totals.csv
    New:        data/exp1/raw_data/distances/ymazeinLT-20230707T140311-data_totals.csv
    New:        data/exp1/raw_data/distances/ymazeinLT-20230707T151509-data_totals.csv
    New:        data/exp1/raw_data/distances/ymazeinLT-20230707T162612-data_totals.csv
    New:        data/exp1/raw_data/zones/ymaze_inLT-20230707T103033.csv
    New:        data/exp1/raw_data/zones/ymaze_inLT-20230707T114209.csv
    New:        data/exp1/raw_data/zones/ymaze_inLT-20230707T125227.csv
    New:        data/exp1/raw_data/zones/ymaze_inLT-20230707T140116.csv
    New:        data/exp1/raw_data/zones/ymaze_inLT-20230707T151154.csv
    New:        data/exp1/raw_data/zones/ymaze_inLT-20230707T162410.csv
    New:        data/exp1/raw_data/zones/ymazeinLT-20230707T103252.csv
    New:        data/exp1/raw_data/zones/ymazeinLT-20230707T114443.csv
    New:        data/exp1/raw_data/zones/ymazeinLT-20230707T125501.csv
    New:        data/exp1/raw_data/zones/ymazeinLT-20230707T140310.csv
    New:        data/exp1/raw_data/zones/ymazeinLT-20230707T151508.csv
    New:        data/exp1/raw_data/zones/ymazeinLT-20230707T162611.csv
    Modified:   data/pilot/14 days/KB 14June2023 melt curve 2_data.xls
    New:        data/pilot/14 days/processed_data/distanceData14days.rds
    New:        data/pilot/14 days/processed_data/final_output.csv
    New:        data/pilot/14 days/processed_data/tidy_data.csv
    New:        data/pilot/14 days/processed_data/time_in_zone.csv
    New:        data/pilot/14 days/raw_data/distances/ymaze_inLT-20230613T103951-data_totals.csv
    New:        data/pilot/14 days/raw_data/distances/ymaze_inLT-20230613T115104-data_totals.csv
    New:        data/pilot/14 days/raw_data/distances/ymaze_inLT-20230613T130222-data_totals.csv
    New:        data/pilot/14 days/raw_data/distances/ymaze_inLT-20230613T141130-data_totals.csv
    New:        data/pilot/14 days/raw_data/distances/ymazeinLT-20230613T104135-data_totals.csv
    New:        data/pilot/14 days/raw_data/distances/ymazeinLT-20230613T115258-data_totals.csv
    New:        data/pilot/14 days/raw_data/distances/ymazeinLT-20230613T130503-data_totals.csv
    New:        data/pilot/14 days/raw_data/distances/ymazeinLT-20230613T141330-data_totals.csv
    New:        data/pilot/14 days/raw_data/zones/ymaze_inLT-20230613T103950.csv
    New:        data/pilot/14 days/raw_data/zones/ymaze_inLT-20230613T115104.csv
    New:        data/pilot/14 days/raw_data/zones/ymaze_inLT-20230613T130221.csv
    New:        data/pilot/14 days/raw_data/zones/ymaze_inLT-20230613T141129.csv
    New:        data/pilot/14 days/raw_data/zones/ymazeinLT-20230613T104134.csv
    New:        data/pilot/14 days/raw_data/zones/ymazeinLT-20230613T115258.csv
    New:        data/pilot/14 days/raw_data/zones/ymazeinLT-20230613T130502.csv
    New:        data/pilot/14 days/raw_data/zones/ymazeinLT-20230613T141329.csv
    New:        data/pilot/14 days/tracking_videos/YMaze_tracking-20230613T104021-f30.avi
    New:        data/pilot/14 days/tracking_videos/YMaze_tracking-20230613T104205-f30.avi
    New:        data/pilot/14 days/tracking_videos/YMaze_tracking-20230613T115135-f30.avi
    New:        data/pilot/14 days/tracking_videos/YMaze_tracking-20230613T115329-f30.avi
    New:        data/pilot/14 days/tracking_videos/YMaze_tracking-20230613T130252-f30.avi
    New:        data/pilot/14 days/tracking_videos/YMaze_tracking-20230613T130533-f30.avi
    New:        data/pilot/14 days/tracking_videos/YMaze_tracking-20230613T141400-f30.avi
    Modified:   data/pilot/7 days/06062023 salbutamol pilot fish metadata.xlsx
    New:        data/pilot/7 days/processed_data/distanceData7days.rds
    New:        output/R_objects/pilot/glmAlts7daysSalbutamol.rds
    New:        output/R_objects/pilot/glmtotalTurnss7daysSalbutamol.rds
    New:        output/R_objects/pilot/salbutamnol_pilot_7days_glmtimeBytotalturns.rds

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with <code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(tidyverse)</code></pre>
<pre><code>── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.2     ✔ readr     2.1.4
✔ forcats   1.0.0     ✔ stringr   1.5.0
✔ ggplot2   3.4.2     ✔ tibble    3.2.1
✔ lubridate 1.9.2     ✔ tidyr     1.3.0
✔ purrr     1.0.1     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
<pre class="r"><code>library(readxl)
library(magrittr)</code></pre>
<pre><code>
Attaching package: &#39;magrittr&#39;

The following object is masked from &#39;package:purrr&#39;:

    set_names

The following object is masked from &#39;package:tidyr&#39;:

    extract</code></pre>
<pre class="r"><code>library(scales)</code></pre>
<pre><code>
Attaching package: &#39;scales&#39;

The following object is masked from &#39;package:purrr&#39;:

    discard

The following object is masked from &#39;package:readr&#39;:

    col_factor</code></pre>
<pre class="r"><code>theme_set(theme_classic())</code></pre>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>In this page, i will analyse the melt curve data generated on the qPCR machine to genotype at the <em>naglu</em> A603 site. In this genotyping assay, I amplify a ~70 bp region over the mutation site in each fish on the qPCR machine. I then perform a melt curva after the PCR amplification is complete.</p>
<p>Since the <em>naglu</em> A603fs mutation deletes 10 bp, the 2 PCR products arising from the mutant and wild type alleles will have a different melting temperature, and this can be observed in the melt curve.</p>
<div id="read-in-data" class="section level2">
<h2>read in data</h2>
<pre class="r"><code># read in the data

# melt cureve deriviities
dfs &lt;- 
  read_delim(&quot;data/exp1.2/genotyping_data/KB_20230821_salbutamolExp1_2_nagluhomxhet7m_genotyping_meltregionderivativedata.txt&quot;, 
             delim = &quot;\t&quot;, # tab delimited
             skip = 8) %&gt;% 
  pivot_longer(names_to = &quot;Reading&quot;,
               values_to = &quot;dF&quot;,
               starts_with(&quot;Reading&quot;)
               ) %&gt;%
  dplyr::select(Position = `Well Location`, Reading, dF)</code></pre>
<pre><code>Rows: 96 Columns: 443
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;\t&quot;
chr   (3): Well Location, Target, Reporter Dye
dbl (440): Well, Reading 1, Reading 2, Reading 3, Reading 4, Reading 5, Read...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<pre class="r"><code># temperature readings
temps &lt;- 
  read_delim(&quot;data/exp1.2/genotyping_data/KB_20230821_salbutamolExp1_2_nagluhomxhet7m_genotyping_meltregiontemperaturedata.txt&quot;,
                    delim = &quot;\t&quot;, 
                    skip = 8
                    ) %&gt;%
  pivot_longer(names_to = &quot;Reading&quot;,
               values_to = &quot;Temperature&quot;,
               starts_with(&quot;Reading&quot;)
               ) %&gt;%
  dplyr::select(Position = `Well Location`, Reading, Temperature)</code></pre>
<pre><code>Rows: 96 Columns: 443
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;\t&quot;
chr   (3): Well Location, Target, Reporter Dye
dbl (440): Well, Reading 1, Reading 2, Reading 3, Reading 4, Reading 5, Read...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<pre class="r"><code># normalised fluorescence from SYBR green data

fluordata &lt;- 
  read_delim(&quot;data/exp1.2/genotyping_data/KB_20230821_salbutamolExp1_2_nagluhomxhet7m_genotyping_meltregionnormalizeddata.txt&quot;,
             delim = &quot;\t&quot;, 
             skip = 8
  ) %&gt;%
  pivot_longer(names_to = &quot;Reading&quot;,
               values_to = &quot;fluor.n&quot;,
               starts_with(&quot;Reading&quot;)
  ) %&gt;%
  dplyr::select(Position = `Well Location`, Reading, fluor.n)</code></pre>
<pre><code>Rows: 96 Columns: 443
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;\t&quot;
chr   (3): Well Location, Target, Reporter Dye
dbl (440): Well, Reading 1, Reading 2, Reading 3, Reading 4, Reading 5, Read...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<pre class="r"><code># results file
data.results &lt;- 
  read_delim(&quot;data/exp1.2/genotyping_data/KB_20230821_salbutamolExp1_2_nagluhomxhet7m_genotyping_results.txt&quot;,
             delim = &quot;\t&quot;, 
             skip = 8
  ) %&gt;%
  dplyr::select(Position = Well, sample = `Sample Name`, Tm1, Tm2, Tm3) %&gt;%
  mutate(group = case_when(
    Position == &quot;A1&quot; ~ &quot;neg control&quot;,
    Position == &quot;A2&quot; ~ &quot;het positive control&quot;,
    Position == &quot;A3&quot; ~ &quot;hom positive control&quot;,
    TRUE ~ &quot;unknown&quot;
  ))</code></pre>
<pre><code>Warning: One or more parsing issues, call `problems()` on your data frame for details,
e.g.:
  dat &lt;- vroom(...)
  problems(dat)</code></pre>
<pre><code>Rows: 100 Columns: 30
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;\t&quot;
chr  (9): Well, Sample Name, Target Name, Task, Reporter, Quencher, Cт, AMPN...
dbl  (8): Cт Mean, Ct Threshold, Baseline Start, Baseline End, Efficiency, T...
lgl (13): RQ, RQ Min, RQ Max, Cт SD, ΔCт, ΔCт Mean, ΔCт SE, HK Control ΔCт M...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<pre class="r"><code># metadata of this fish and ckeanup
meta &lt;- read_excel(&quot;data/exp1.2/2023_Aug18_salbutamolexp1_2_meta.xlsx&quot;) %&gt;%
  mutate(fish_id = as.character(fish_id),
         
         treatment = factor(treatment,
                            levels = c(&quot;untreated&quot;, &quot;20 µM Salbutamol&quot;)),
         
         sex = as.factor(sex),
         
         behavBatch = as.factor(behavBatch),
         
         ymazeUnit = as.factor(ymazeUnit),
         
         ymazePosition = as.factor(ymazePosition)) %&gt;%
  
  dplyr::select(fish_id, treatment, sex, Position = PositionForGeno, everything())



# join these together
data.meltcurve &lt;-
  dfs %&gt;%
  left_join(temps) %&gt;% 
  left_join(fluordata) </code></pre>
<pre><code>Joining with `by = join_by(Position, Reading)`
Joining with `by = join_by(Position, Reading)`</code></pre>
</div>
</div>
<div id="plot-out-fluorescence-data" class="section level1">
<h1>Plot out fluorescence data</h1>
<p>As the temp increases, the fluorescence decreaes. This is consistent with theee PCR products melting and the SYBR Green no longer fluorescing.</p>
<pre class="r"><code>data.meltcurve %&gt;%
  left_join(data.results) %&gt;%
  ggplot(aes(x = Temperature, y = fluor.n), 
         alpha = 0.75) +
  geom_line(aes(group = Position,
                colour = group)) +
  scale_x_continuous(limits = c(78,86),
                     breaks = seq(70,90)) +
  scale_y_continuous(labels = comma)</code></pre>
<pre><code>Joining with `by = join_by(Position)`</code></pre>
<pre><code>Warning: Removed 32264 rows containing missing values (`geom_line()`).</code></pre>
<p><img src="figure/exp1.2_meltCurveGenotyping.rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="plot-out-the-first-derivitive-of-the-fluorescence-data" class="section level1">
<h1>Plot out the first derivitive of the fluorescence data</h1>
<p>Two clear peaks are observed consistent with the wt and mutant alleles being present.</p>
<pre class="r"><code>data.meltcurve %&gt;%
  left_join(data.results) %&gt;%
  ggplot(aes(x = Temperature, y = dF)) +
  geom_line(aes(group = Position,
                size = group,
                alpha = group, 
                colour = group)) +
  scale_size_manual(values = c(2, 2, 2, 0.5)) + # make the controls fatter to stand out more
  scale_alpha_manual(values = c(1, 1, 1, 0.5)) + # make the unknowns more transparent so the controls stand out more. 
  scale_x_continuous(limits = c(78,86),
                     breaks = seq(70,90)) +
  scale_y_continuous(labels = comma, 
                     limits = c(0, 300000))</code></pre>
<pre><code>Joining with `by = join_by(Position)`</code></pre>
<pre><code>Warning: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.
ℹ Please use `linewidth` instead.
This warning is displayed once every 8 hours.
Call `lifecycle::last_lifecycle_warnings()` to see where this warning was
generated.</code></pre>
<pre><code>Warning: Removed 32264 rows containing missing values (`geom_line()`).</code></pre>
<p><img src="figure/exp1.2_meltCurveGenotyping.rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="define-the-hets-and-homs" class="section level1">
<h1>define the hets and homs</h1>
<pre class="r"><code>data.results &lt;- data.results %&gt;%
  mutate(wtallele = case_when(
    Tm1 &gt; 81.9 ~ TRUE,  #if the melt temp is higher than 83C, this is the wt band and so het
    TRUE ~ FALSE
  ),
  mutallele = case_when(
    Tm1 &lt; 81.9  ~ TRUE,
    TRUE ~ FALSE
  ),
  inferredGenotype = case_when(
    Position == &quot;A1&quot; ~ &quot;neg control&quot;,
    Position == &quot;A2&quot; ~ &quot;het positive control&quot;,
    Position == &quot;A3&quot; ~ &quot;hom positive control&quot;,
     wtallele ==  TRUE &amp; mutallele == FALSE ~ &quot;het&quot;,
    wtallele ==  TRUE &amp; mutallele == TRUE ~ &quot;het&quot;,
    wtallele ==  FALSE &amp; mutallele == TRUE ~ &quot;hom&quot;
  )
  )</code></pre>
</div>
<div id="plot-out-the-results-by-genotype" class="section level1">
<h1>plot out the results by genotype</h1>
<pre class="r"><code># repeat the plot
data.meltcurve %&gt;%
  left_join(data.results) %&gt;%
  ggplot(aes(x = Temperature, y = dF)) +
  geom_line(aes(group = Position,
                size = group,
                alpha = group, 
                colour = inferredGenotype)) +
  scale_size_manual(values = c(2, 2, 2, 0.5)) + # make the controls fatter to stand out more
  scale_alpha_manual(values = c(1, 1, 1, 0.5)) + # make the unknowns more transparent so the controls
  scale_x_continuous(limits = c(78,86),
                     breaks = seq(70,90)) +
  scale_y_continuous(labels = comma, 
                     limits = c(0, 300000))</code></pre>
<pre><code>Joining with `by = join_by(Position)`</code></pre>
<pre><code>Warning: Removed 32264 rows containing missing values (`geom_line()`).</code></pre>
<p><img src="figure/exp1.2_meltCurveGenotyping.rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="export-the-data" class="section level1">
<h1>export the data</h1>
<pre class="r"><code># export data for the final document
meta %&gt;%
  left_join(data.results %&gt;%
              dplyr::select(Position, genotype = inferredGenotype)) %&gt;%
  dplyr::select(1,2,3, genotype, everything()) %&gt;%
  saveRDS(&quot;data/exp1.2/metadata_withGenotype.rds&quot;)</code></pre>
<pre><code>Joining with `by = join_by(Position)`</code></pre>
<pre class="r"><code>data.meltcurve %&gt;%
  left_join(data.results) %&gt;%
  saveRDS(&quot;data/exp1.2/genotyping_data/meltcurveWithGenotypes.rds&quot;)</code></pre>
<pre><code>Joining with `by = join_by(Position)`</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.3 (2023-03-15)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Catalina 10.15.7

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib

locale:
[1] en_AU.UTF-8/en_AU.UTF-8/en_AU.UTF-8/C/en_AU.UTF-8/en_AU.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] scales_1.2.1    magrittr_2.0.3  readxl_1.4.2    lubridate_1.9.2
 [5] forcats_1.0.0   stringr_1.5.0   dplyr_1.1.2     purrr_1.0.1    
 [9] readr_2.1.4     tidyr_1.3.0     tibble_3.2.1    ggplot2_3.4.2  
[13] tidyverse_2.0.0

loaded via a namespace (and not attached):
 [1] tidyselect_1.2.0 xfun_0.39        bslib_0.4.2      colorspace_2.1-0
 [5] vctrs_0.6.2      generics_0.1.3   htmltools_0.5.5  yaml_2.3.7      
 [9] utf8_1.2.3       rlang_1.1.1      jquerylib_0.1.4  later_1.3.1     
[13] pillar_1.9.0     glue_1.6.2       withr_2.5.0      bit64_4.0.5     
[17] lifecycle_1.0.3  cellranger_1.1.0 munsell_0.5.0    gtable_0.3.3    
[21] workflowr_1.7.0  evaluate_0.21    labeling_0.4.2   knitr_1.42      
[25] tzdb_0.3.0       fastmap_1.1.1    httpuv_1.6.10    parallel_4.2.3  
[29] fansi_1.0.4      highr_0.10       Rcpp_1.0.10      promises_1.2.0.1
[33] cachem_1.0.8     vroom_1.6.3      jsonlite_1.8.4   farver_2.1.1    
[37] bit_4.0.5        fs_1.6.2         hms_1.1.3        digest_0.6.31   
[41] stringi_1.7.12   grid_4.2.3       rprojroot_2.0.3  cli_3.6.1       
[45] tools_4.2.3      sass_0.4.6       crayon_1.5.2     pkgconfig_2.0.3 
[49] timechange_0.2.0 rmarkdown_2.21   rstudioapi_0.14  R6_2.5.1        
[53] git2r_0.32.0     compiler_4.2.3  </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
